*** ./crypto/o_str.c	2026-06-09 19:54:25.000000000 +0800
--- /tmp/o_str.c	2026-09-27 12:14:34.942635351 +0800
***************
*** 373,382 ****
--- 373,383 ----
  int openssl_strerror_r(int errnum, char *buf, size_t buflen)
  {
  #if defined(_MSC_VER) && _MSC_VER >= 1400 && !defined(_WIN32_WCE)
      return !strerror_s(buf, buflen, errnum);
  #elif defined(_GNU_SOURCE)
+ #ifdef __GLIBC__
      char *err;
  
      /*
       * GNU strerror_r may not actually set buf.
       * It can return a pointer to some (immutable) static string in which case
***************
*** 392,401 ****
--- 393,405 ----
       * reads src byte for byte and writes to dest.
       * If err == buf we do not have to copy anything.
       */
      if (err != buf)
          OPENSSL_strlcpy(buf, err, buflen);
+ #else /* For musl library, `strerror_r returns an int */
+     strerror_r(errnum, buf, buflen);
+ #endif
      return 1;
  #elif (defined(_POSIX_C_SOURCE) && _POSIX_C_SOURCE >= 200112L) || (defined(_XOPEN_SOURCE) && _XOPEN_SOURCE >= 600)
      /*
       * We can use "real" strerror_r. The OpenSSL version differs in that it
       * gives 1 on success and 0 on failure for consistency with other OpenSSL
