diff --git a/Makefile b/Makefile
index 1bffcae..ca458af 100644
--- a/Makefile
+++ b/Makefile
@@ -9,16 +9,15 @@ PLAT= none
 
 # Where to install. The installation starts in the src and doc directories,
 # so take care if INSTALL_TOP is not an absolute path.
-INSTALL_TOP= /usr/local
+INSTALL_TOP ?= /usr/local
 INSTALL_BIN= $(INSTALL_TOP)/bin
 INSTALL_INC= $(INSTALL_TOP)/include
 INSTALL_LIB= $(INSTALL_TOP)/lib
-INSTALL_MAN= $(INSTALL_TOP)/man/man1
 #
 # You probably want to make INSTALL_LMOD and INSTALL_CMOD consistent with
 # LUA_ROOT, LUA_LDIR, and LUA_CDIR in luaconf.h (and also with etc/lua.pc).
-INSTALL_LMOD= $(INSTALL_TOP)/share/lua/$V
-INSTALL_CMOD= $(INSTALL_TOP)/lib/lua/$V
+INSTALL_LMOD= $(INSTALL_TOP)/share/lua
+INSTALL_CMOD= $(INSTALL_TOP)/lib/lua
 
 # How to install. If your install program does not support "-p", then you
 # may have to run ranlib on the installed liblua.a (do "make ranlib").
@@ -41,10 +40,9 @@ RANLIB= ranlib
 PLATS= aix ansi bsd freebsd generic linux macosx mingw posix solaris
 
 # What to install.
-TO_BIN= lua$V luac$V
+TO_BIN= lua luac
 TO_INC= lua.h luaconf.h lualib.h lauxlib.h ../etc/lua.hpp lnum_config.h
-TO_LIB= liblua.a liblua.so.$R
-TO_MAN= lua$V.1 luac$V.1
+TO_LIB= liblua.so.$R
 
 # Lua version and release.
 V= 5.1
@@ -59,12 +57,11 @@ test:	dummy
 	src/lua test/hello.lua
 
 install: dummy
-	cd src && $(MKDIR) $(INSTALL_BIN) $(INSTALL_INC) $(INSTALL_LIB) $(INSTALL_MAN) $(INSTALL_LMOD) $(INSTALL_CMOD)
+	cd src && $(MKDIR) $(INSTALL_BIN) $(INSTALL_INC) $(INSTALL_LIB) $(INSTALL_LMOD) $(INSTALL_CMOD)
 	cd src && $(INSTALL_EXEC) $(TO_BIN) $(INSTALL_BIN)
 	cd src && $(INSTALL_DATA) $(TO_INC) $(INSTALL_INC)
 	cd src && $(INSTALL_DATA) $(TO_LIB) $(INSTALL_LIB)
-	ln -s liblua.so.$R $(INSTALL_LIB)/liblua.so
-	cd doc && $(INSTALL_DATA) $(TO_MAN) $(INSTALL_MAN)
+	rm -f -v $(INSTALL_LIB)/liblua.so && ln -sv liblua.so.$R $(INSTALL_LIB)/liblua.so
 
 ranlib:
 	cd src && cd $(INSTALL_LIB) && $(RANLIB) $(TO_LIB)
@@ -96,7 +93,6 @@ echo:
 	@echo "INSTALL_BIN = $(INSTALL_BIN)"
 	@echo "INSTALL_INC = $(INSTALL_INC)"
 	@echo "INSTALL_LIB = $(INSTALL_LIB)"
-	@echo "INSTALL_MAN = $(INSTALL_MAN)"
 	@echo "INSTALL_LMOD = $(INSTALL_LMOD)"
 	@echo "INSTALL_CMOD = $(INSTALL_CMOD)"
 	@echo "INSTALL_EXEC = $(INSTALL_EXEC)"
@@ -112,7 +108,6 @@ pecho:
 	@echo "TO_BIN = $(TO_BIN)"
 	@echo "TO_INC = $(TO_INC)"
 	@echo "TO_LIB = $(TO_LIB)"
-	@echo "TO_MAN = $(TO_MAN)"
 
 # echo config parameters as Lua code
 # uncomment the last sed expression if you want nil instead of empty strings
diff --git a/src/Makefile b/src/Makefile
index f0d327f..1bc69fc 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -5,25 +5,25 @@
 # == CHANGE THE SETTINGS BELOW TO SUIT YOUR ENVIRONMENT =======================
 
 # Your platform. See PLATS for possible values.
-PLAT= none
+PLAT ?= none
 
-CC= gcc
-CFLAGS= -O2 -Wall $(MYCFLAGS)
-AR= ar rcu
-RANLIB= ranlib
+CC = $(CROSS_COMPILE)gcc
+CFLAGS = -O2 -Wall $(MYCFLAGS) $(EXTRA_CFLAGS)
+AR= $(CROSS_COMPILE)ar rcu
+RANLIB= $(CROSS_COMPILE)ranlib
 RM= rm -f
 LIBS= -lm $(MYLIBS)
 
 MYCFLAGS=
 MYLDFLAGS=
 MYLIBS=
+PKG_VERSION=5.1.5
 # USE_READLINE=1
 
 # == END OF USER SETTINGS. NO NEED TO CHANGE ANYTHING BELOW THIS LINE =========
 
 PLATS= aix ansi bsd freebsd generic linux macosx mingw posix solaris
 
-LUA_A=	liblua.a
 LUA_SO= liblua.so
 CORE_O=	lapi.o lcode.o ldebug.o ldo.o ldump.o lfunc.o lgc.o llex.o lmem.o \
 	lobject.o lopcodes.o lparser.o lstate.o lstring.o ltable.o ltm.o  \
@@ -31,15 +31,14 @@ CORE_O=	lapi.o lcode.o ldebug.o ldo.o ldump.o lfunc.o lgc.o llex.o lmem.o \
 LIB_O=	lauxlib.o lbaselib.o ldblib.o liolib.o lmathlib.o loslib.o ltablib.o \
 	lstrlib.o loadlib.o linit.o
 
-LUA_T=	lua$V
+LUA_T=	lua
 LUA_O=	lua.o
 
-LUAC_T=	luac$V
+LUAC_T=	luac
 LUAC_O=	luac.o print.o lopcodes.o
 
 ALL_O= $(CORE_O) $(LIB_O) $(LUA_O) $(LUAC_O)
-ALL_T= $(LUA_A) $(LUA_SO) $(LUA_T) $(LUAC_T)
-ALL_A= $(LUA_A)
+ALL_T= $(LUA_SO) $(LUA_T) $(LUAC_T)
 ALL_SO= $(LUA_SO)
 
 default: $(PLAT)
@@ -48,29 +47,23 @@ all:	$(ALL_T)
 
 o:	$(ALL_O)
 
-a:	$(ALL_A)
-
 so:	$(ALL_SO)
 
-$(LUA_A): $(CORE_O) $(LIB_O)
-	$(AR) $@ $(CORE_O) $(LIB_O)	# DLL needs all object files
-	$(RANLIB) $@
-
 $(LUA_SO): $(CORE_O) $(LIB_O)
 	$(CC) -o $@.$(PKG_VERSION) -Wl,-Bsymbolic-functions $(MYLDFLAGS) -shared -Wl,-soname="$@.$(PKG_VERSION)" $?
 	ln -fs $@.$(PKG_VERSION) $@
 
 $(LUA_T): $(LUA_O) $(LUA_SO)
-	$(CC) -o $@ -L. -llua $(MYLDFLAGS) $(LUA_O) $(LIBS)
+	$(CC) -o $@ $(MYLDFLAGS) $(LUA_O) -L. -llua $(LIBS) -Wl,-rpath=$(INSTALL_TOP)/lib
 
 $(LUAC_T): $(LUAC_O) $(LUA_SO)
-	$(CC) -o $@ -L. -llua $(MYLDFLAGS) $(LUAC_O) $(LIBS)
+	$(CC) -o $@ $(MYLDFLAGS) $(LUAC_O) -L. -llua $(LIBS) -Wl,-rpath=$(INSTALL_TOP)/lib
 
 $(LUAC_T)-host: $(LUAC_O) $(LUA_A)
 	$(CC) -o $@ $(MYLDFLAGS) $(LUAC_O) $(LUA_A) $(LIBS)
 
 clean:
-	$(RM) $(ALL_T) $(ALL_O)
+	$(RM) $(ALL_T) $(ALL_O) *.so.$(PKG_VERSION)
 
 depend:
 	@$(CC) $(CFLAGS) -MM l*.c print.c
diff --git a/src/lnum_config.h b/src/lnum_config.h
index 1092eea..bdeac93 100644
--- a/src/lnum_config.h
+++ b/src/lnum_config.h
@@ -10,11 +10,10 @@
 /*
 ** Default number modes
 */
-#if (!defined LNUM_DOUBLE) && (!defined LNUM_FLOAT) && (!defined LNUM_LDOUBLE)
-# define LNUM_DOUBLE
-#endif
-#if (!defined LNUM_INT16) && (!defined LNUM_INT32) && (!defined LNUM_INT64)
-# define LNUM_INT32
+#define LNUM_DOUBLE 1
+#define LNUM_INT64  1
+#ifdef LNUM_INT32
+  #undef LNUM_INT32
 #endif
 
 /*
diff --git a/src/luaconf.h b/src/luaconf.h
index e20053e..ec64957 100644
--- a/src/luaconf.h
+++ b/src/luaconf.h
@@ -95,7 +95,9 @@
 	".\\?.dll;"  LUA_CDIR"?.dll;" LUA_CDIR"loadall.dll"
 
 #else
-#define LUA_ROOT	"/usr/"
+#ifndef LUA_ROOT
+  #define LUA_ROOT	"/usr/"
+#endif
 #define LUA_LDIR	LUA_ROOT "share/lua/"
 #define LUA_CDIR	LUA_ROOT "lib/lua/"
 #define LUA_PATH_DEFAULT  \
@@ -497,13 +499,13 @@
 @@ LUAI_MAXUPVALUES is the maximum number of upvalues per function
 @* (must be smaller than 250).
 */
-#define LUAI_MAXUPVALUES	60
+#define LUAI_MAXUPVALUES	200
 
 
 /*
 @@ LUAL_BUFFERSIZE is the buffer size used by the lauxlib buffer system.
 */
-#define LUAL_BUFFERSIZE		BUFSIZ
+#define LUAL_BUFFERSIZE		8192
 
 /* }================================================================== */
 
