diff --git a/delta.c b/delta.c
index 5437fc1..71f4e3d 100644
--- a/delta.c
+++ b/delta.c
@@ -15,8 +15,9 @@
 /*
  * This file contains the code for handling uci config delta files
  */
-
-#define _GNU_SOURCE
+#ifndef _GNU_SOURCE
+  #define _GNU_SOURCE
+#endif
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <sys/file.h>
diff --git a/file.c b/file.c
index 6840cfc..3ea42d9 100644
--- a/file.c
+++ b/file.c
@@ -16,7 +16,9 @@
  * This file contains the code for parsing uci config files
  */
 
-#define _GNU_SOURCE
+#ifndef _GNU_SOURCE
+  #define _GNU_SOURCE
+#endif
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <sys/file.h>
diff --git a/libuci.c b/libuci.c
index d854549..7e160eb 100644
--- a/libuci.c
+++ b/libuci.c
@@ -15,8 +15,9 @@
 /*
  * This file contains some common code for the uci library
  */
-
-#define _GNU_SOURCE
+#ifndef _GNU_SOURCE
+  #define _GNU_SOURCE
+#endif
 #include <sys/types.h>
 #include <stdbool.h>
 #include <string.h>
diff --git a/util.c b/util.c
index 61e42cd..588d2f1 100644
--- a/util.c
+++ b/util.c
@@ -16,7 +16,9 @@
  * This file contains misc utility functions and wrappers to standard
  * functions, which throw exceptions upon failure.
  */
-#define _GNU_SOURCE
+#ifndef _GNU_SOURCE
+  #define _GNU_SOURCE
+#endif
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <sys/file.h>
